internal/controllers/machinehealthcheck/machinehealthcheck_controller_test.go

```diff
@@ -405,14 +405,26 @@ func TestMachineHealthCheck_Reconcile(t *testing.T) {
 		)
 		defer cleanup2()
 		// Unhealthy nodes and machines but already in deletion.
+		// Note: deletionTimestamp gets set by deletion below which also removes the skip remediation annotation.
 		_, unhealthyMachinesDeleting, cleanup3 := createMachinesWithNodes(g, cluster,
 			count(1),
 			createNodeRefForMachine(true),
 			nodeStatus(corev1.ConditionUnknown),
 			machineLabels(mhc.Spec.Selector.MatchLabels),
-			machineDeleting(),
+			machineFinalizers("test.cluster.io/delete-protection"),
```

Comment:
q: when are dropping the finalizer?

Reply:
No, we need it otherwise the object is gone.